File: src/main/java/org/polyfrost/example/MageAddonsMod.java
package org.polyfrost.example;

import net.minecraft.client.Minecraft;
import net.minecraft.client.settings.KeyBinding;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.InputEvent;
import org.lwjgl.input.Keyboard;
import org.polyfrost.example.command.MACommand;
import org.polyfrost.example.config.MAConfig;
import cc.polyfrost.oneconfig.events.event.InitializationEvent;
import net.minecraftforge.fml.common.Mod;
import cc.polyfrost.oneconfig.utils.commands.CommandManager;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.client.registry.ClientRegistry;
import net.minecraftforge.fml.common.gameevent.InputEvent;

/**
 * The entrypoint of the Example Mod that initializes it.
 *
 * @see Mod
 * @see InitializationEvent
 */
@Mod(modid = MageAddonsMod.MODID, name = MageAddonsMod.NAME, version = MageAddonsMod.VERSION)
public class MageAddonsMod {
    // Sets the variables from `gradle.properties`. See the `blossom` config in `build.gradle.kts`.
    public static final String MODID = "@ID@";
    public static final String NAME = "@NAME@";
    public static final String VERSION = "@VER@";
    @Mod.Instance(MODID)
    public static MageAddonsMod INSTANCE; // Adds the instance of the mod, so we can access other variables.
    public static MAConfig config;

    static KeyBinding tarKey1;

    // Register the config and commands.
    @Mod.EventHandler
    public void onInit(FMLInitializationEvent event) {
        config = new MAConfig();
        CommandManager.INSTANCE.registerCommand(new MACommand());

        int keyCode = config.getKeyCode();
        tarKey1 = new KeyBinding("/eq command", Keyboard.KEY_X, "null");
        ClientRegistry.registerKeyBinding(tarKey1);
    }

    // Listen for key input events
    @SubscribeEvent
    public void onKeyInput(InputEvent.KeyInputEvent event) {
        if (tarKey1.isPressed()) {
            Minecraft.getMinecraft().thePlayer.sendChatMessage("sent1");
        }
    }
}

File: src/main/java/ultimate/huh/kotlin/MageAddons/MageAddonsMod.kt
package ultimate.huh.kotlin.MageAddons

import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import java.io.File

class MageAddonsMod {

    @Mod(
        modid = MageAddonsMod.MOD_ID,
        name = MageAddonsMod.MOD_NAME,
        version = MageAddonsMod.MOD_VERSION,
        clientSideOnly = true
    )
    class MageAddonsMod{
        @Mod.EventHandler
        fun preInit(event: FMLPreInitializationEvent) {
            val directory = File(event.modConfigurationDirectory, "mageaddonsmod")
            directory.mkdirs()
        }
        @Mod.EventHandler
        fun onInit(event: FMLInitializationEvent) {
        }



        companion object {
            const val MOD_ID = "mageaddons"
            const val MOD_NAME = "Mage Addons"
            const val MOD_VERSION = "0.0.1"
            const val CHAT_PREFIX = "§f<§3MageAddons§f>§r"

            val mc: Minecraft = Minecraft.getMinecraft()
            var display: GuiScreen? = null


        }

    }
}